### PR TITLE
BE: Adds gmail auth route

### PR DESCRIPTION
**Pull request by cpascale43**
Re #12.

**What it does**
- Adds Gmail auth route, OAuth authentication methods from Google and tests

**What to look for**
- This PR hands off backend work on Gmail auth to @Patton-L

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+name = "pypi"
+url = "https://pypi.org/simple"
+verify_ssl = true
+
+[dev-packages]
+
+[packages]
+
+[requires]
+python_version = "3.9"
```
Comment (skassam21) [inline Pipfile:11]: Can we remove the pipfiles in the root directory? We don't need them

**Review by skassam21 — COMMENTED** [review]
Do you know why there is overlapping file changes with a lot of other PRs? Maybe it may be easier to keep this open for @Patton-L  to see what you did, but work off a new branch for him to continuing working on the feature? I'll let you both decide the best way forward for this

**Comment by Patton-L** [comment]
> Do you know why there is overlapping file changes with a lot of other PRs? Maybe it may be easier to keep this open for @Patton-L to see what you did, but work off a new branch for him to continuing working on the feature? I'll let you both decide the best way forward for this

Will rebase solve this problem?

File: Pipfile (hunk 1 above)
Comment (cpascale43) [reply Pipfile:11]: Closing because opened new PR #44

**Comment by cpascale43** [comment]
Closing this because opened new PR #44

**Pull request closed without merging**